todocore/src/main/java/com/knighttodo/todocore/service/RoutineTodoInstanceService.java

```diff
@@ -3,36 +3,41 @@
 import com.knighttodo.todocore.domain.RoutineTodoInstanceVO;
 import com.knighttodo.todocore.domain.RoutineTodoVO;
 import com.knighttodo.todocore.exception.RoutineTodoNotFoundException;
-import com.knighttodo.todocore.gateway.RoutineTodoInstanceGateway;
-import com.knighttodo.todocore.gateway.character.ExperienceGateway;
+import com.knighttodo.todocore.service.character.ExperienceServ;
+import com.knighttodo.todocore.service.privatedb.mapper.RoutineTodoInstanceMapper;
+import com.knighttodo.todocore.service.privatedb.repository.RoutineTodoInstanceRepository;
 import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
 import org.springframework.stereotype.Service;
 import org.springframework.transaction.annotation.Transactional;
 
 import java.util.List;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 @Service
 @Slf4j
 @RequiredArgsConstructor
 @Transactional(readOnly = true)
 public class RoutineTodoInstanceService {
 
-    private final RoutineTodoInstanceGateway routineTodoInstanceGateway;
     private final RoutineInstanceService routineInstanceService;
-    private final ExperienceGateway experienceGateway;
+    private final ExperienceServ experienceService;
+    private final RoutineTodoInstanceRepository routineTodoInstanceRepository;
+    private final RoutineTodoInstanceMapper routineTodoInstanceMapper;
 
     @Transactional
     public RoutineTodoInstanceVO save(UUID routineInstanceId, RoutineTodoInstanceVO routineTodoInstanceVO) {
         routineTodoInstanceVO.setRoutineInstanceVO(routineInstanceService.findById(routineInstanceId));
-        RoutineTodoInstanceVO savedRoutineTodoInstance = routineTodoInstanceGateway.save(routineTodoInstanceVO);
+        RoutineTodoInstanceVO savedRoutineTodoInstance =
+                routineTodoInstanceMapper.toRoutineTodoInstanceVO(routineTodoInstanceRepository.save(routineTodoInstanceMapper.toRoutineTodoInstance(routineTodoInstanceVO)));
```

Comment:
it looks too long for me, can you move the part of this line down?

Reply:
Yes. It's too long...